### DISCUSSION TITLE
subdirectory

### DISCUSSION THREAD

**Discussion by jcgmalave** [filamentphp/filament#1918] [Help]
The problem is the following: we have developed a Laravel application and we have to deploy it on a server that has another non-Laravel application, a landing page, or a "normal" web page, for example, in the root directory.

Suppose the page is on an intranet, and has the name https://mycorp.com.

In some corporate environments, we do not have control over DNS, or there are policies that do not allow the creation of subdomains, so we cannot create a virtual server in Apache, which would solve our problems by creating something like https:/ /myapp.mycorp.com

Due to the above, we will have to install the Laravel application under a subdirectory of the main server, so that it remains https://micorp.com/miapp.

when doing this, filament stops working, and I can't get past the login.

**Comment by danharrin** [comment]
Have you updated the `path` in the config file? Maybe that makes a difference.

**Reply by jcgmalave** [reply]
I did, but in the filament.php teresa múltiple path variables, which one should I update?

**Comment by PrasadChinwal** [comment]
Has anyone figured out the solution for this? 
I am using Filament 3.x and my base url is `https://myapp.com/app`. I have configured my Filament provider as below:
```php
public function panel(Panel $panel): Panel
    {
        return $panel
            ->default()
            ->id('CampusRec')
            ->path('/filament')
            ->login()
            ->colors([
                'primary' => Color::Amber,
            ])
            ->discoverResources(in: app_path('Filament/Resources'), for: 'App\\Filament\\Resources')
            ->discoverPages(in: app_path('Filament/Pages'), for: 'App\\Filament\\Pages')
            ->pages([
                Pages\Dashboard::class,
            ])
            ->discoverWidgets(in: app_path('Filament/Widgets'), for: 'App\\Filament\\Widgets')
            ->widgets([
                Widgets\AccountWidget::class,
                Widgets\FilamentInfoWidget::class,
            ])
            ->middleware([
                EncryptCookies::class,
                AddQueuedCookiesToResponse::class,
                StartSession::class,
                AuthenticateSession::class,
                ShareErrorsFromSession::class,
                VerifyCsrfToken::class,
                SubstituteBindings::class,
                DisableBladeIconComponents::class,
                DispatchServingFilamentEvent::class,
            ])
            ->authMiddleware([
                Authenticate::class,
            ]);
    }
```

I am able to visit `https://myapp.com/app/filament` and view the dashboard but nothing is working. I can't even click any buttons. 
I have no errors in console too.

**Reply by resoftw** [reply]
I have the same issue too..
the problem is the livewire paths
and I haven't got any solution yet..

**Comment by resoftw** [comment]
It's livewire problem..

**Reply by leandrocfe** [reply]
console errors?

**Comment by undershell** [comment]
I have already tested redeploying a Filament project in a subdirectory, and it worked successfully.

First, you need to deploy your Laravel project as you usually do. You can keep your Laravel project outside of the web directory and create a symbolic link to the `public/` directory, or copy it and redefine the desired path in `index.php`, or even customize a virtual host for the subdirectory with a root directory (e.g. DocumentRoot) pointing to the `public/` folder.

After changing your `APP_URL` in the `.env` or config file:

```dotenv
APP_URL=https://example.com/myproject
```

You may notice that some resources are not loading in Filament, such as Livewire. By default, Livewire assets are served from the route `/livewire/livewire.js`. However, if your application is hosted on a non-root path of your domain, such as `example.com/myproject`, you will need to configure the Livewire `asset_url` setting found in `config/livewire.php`:

```php
'asset_url' => null,
```

You can redefine it as:

```php
'asset_url' => '/myproject',
// or
'asset_url' => env('APP_URL'),
```

Additionally, if you have any hard-coded URLs inside the views, you need to update them. This also applies to custom or compiled assets. For instance, if you customized paths through Laravel Mix, you may need to verify and update the URLs in your `/public/mix-manifest.json` file accordingly:

```json
{
    "/js/app.js": "/myproject/js/app.js",
    //…
}
```

Clear your configuration and cache to apply the changes, and optionally rebuild them:

```bash
php artisan config:clear
php artisan cache:clear
php artisan view:clear

php artisan config:cache
php artisan route:cache
php artisan cache:clear
```
